sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -557,6 +557,20 @@ abstract class JdbcDialect extends Serializable with Logging {
   def getJdbcSQLQueryBuilder(options: JDBCOptions): JdbcSQLQueryBuilder =
     new JdbcSQLQueryBuilder(this, options)
 
+  /**
+   * Returns ture if dialect supports LIMIT clause.
+   *
+   * Note: Some build-in dialect supports LIMIT clause with some trick, please see:
+   * {@link OracleDialect.OracleSQLQueryBuilder} and
+   * {@link MsSqlServerDialect.MsSqlServerSQLQueryBuilder}.
+   */
+  def supportsLimit: Boolean = true
+
+  /**
+   * Returns ture if dialect supports OFFSET clause.
+   */
+  def supportsOffset: Boolean = true
```

Comment:
Do all the dialects support offset? If not, do we need to set this to false in the dialect?

BTW you still have this `TODO[SPARK-42289]: DS V2 pushdown could let JDBC dialect decide to push down offset` in `OracleDialect` and `MsSqlServerDialect`

Reply:
Most databases can directly or indirectly support offset. so I think we should keep true here.

Reply:
> BTW you still have this `TODO[SPARK-42289]: DS V2 pushdown could let JDBC dialect decide to push down offset` in `OracleDialect` and `MsSqlServerDialect`

Yes. It need another PR. This PR just open the API.

Reply:
I think it's better to use false as the default value, and override it in builtin dialects in return true.

Reply:
given that we will turn on the JDBC option to enable pushdown soon.

Reply:
We can do it in the PR switching the default value of JDBC pushdown options.